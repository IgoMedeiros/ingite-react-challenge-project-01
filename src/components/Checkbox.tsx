import styles from "./Checkbox.module.css";

export function Checkbox({ checked }: { checked: boolean }) {
  return (
    <input type="checkbox" className={styles.checkbox} checked={checked} />
  );
}
