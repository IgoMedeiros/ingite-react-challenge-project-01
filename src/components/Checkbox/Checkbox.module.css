.checkbox[type='checkbox'] {
  width: 21px;
  height: 21px;
  cursor: pointer;
  appearance: none;
  -webkit-appearance: none;
  border-radius: 999px;
  border: 2px solid var(--blue);
}

.checkbox[type='checkbox']:hover {
  border-color: var(--blue-dark);
}

.checkbox[type='checkbox']:checked {
  border-color: var(--purple-dark);
  background-color: var(--purple-dark);
  display: flex;
  align-items: center;
  justify-content: center;
}

.checkbox[type='checkbox']:checked::before {
  content: url(../../assets/checked_vector.svg);
  color: var(--gray-100);
  display: flex;
  width: 0.75rem;
  height: 0.75rem;
}

.checkbox[type='checkbox']:checked:hover {
  border-color: var(--purple);
  background-color: var(--purple);
}

.checkbox[type='checkbox']:focus {
  box-shadow: none;
}
