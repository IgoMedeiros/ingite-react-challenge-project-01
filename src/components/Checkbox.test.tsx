import { render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import { Checkbox } from './Checkbox'

describe('Checkbox', () => {
  it('should render', () => {
    const wrapper = render(<Checkbox checked={false} />)
    expect(wrapper.container).toMatchSnapshot()
  })
})
