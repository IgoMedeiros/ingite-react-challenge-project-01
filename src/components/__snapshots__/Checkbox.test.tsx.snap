// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox > should render 1`] = `
<div>
  <input
    class="_checkbox_f7bd51"
    type="checkbox"
  />
</div>
`;
